### ISSUE TITLE
Can I use the Sandbox Endpoint from api.sap.com

### ISSUE THREAD

**Issue by gregorwolf** [jungsap/capWithS4#2]
Hi Thomas,

in [cat-service.js#L22](https://github.com/jungsap/capWithS4/blob/master/srv/cat-service.js#L22) I see the hardcoded URL of the API endpoint. Would it be possible to replace that with the Sandbox offered for the Service:

https://api.sap.com/api/API_BUSINESS_PARTNER/resource

?

Best regards
Gregor

**Comment by jungsap** [comment]
The problem I see with this option is that the API hub sandbox requires an apikey header in all requests. I don't see any option in the S/4 Cloud SDK JavaScript to add the apikey.  I checked in the cloud-sdk-core destination.js and it seems only setup to take in URL, Name, User, Password, ProxyType, sapClient, Authentication, and authTokens.  There is an originalProperties that might store other attributes. However if you looks at the request header builder (header-builder.js) there is a mention of custom headers. So now I have some hope:

 * Create object containing all headers, including custom headers for a given  OData request configuration and destination.
 * Custom headers override duplicate headers.

I finally found in the GetAllRequestBuilder docu that there is an option to use withCustomHeaders and through this I passed the API key. It worked well then.  I've add those lines commented out and pushed the additional code.  
<img width="601" alt="2019-03-29_8-22-07" src="https://user-images.githubusercontent.com/12159356/55232382-c7120980-51fb-11e9-984f-9bf7438fdac4.png">
